Modules/IO/NIFTI/src/itkNiftiImageIO.cxx

```diff
@@ -521,6 +523,36 @@ CastCopy(float * to, const void * from, size_t pixelcount)
   }
 }
 
+// Internal function to convert vectors between RAS and LPS coordinate systems.
+// Dimensions are CXYZT (ITK memory layout)
+template <typename TBuffer>
+void
+ConvertRASToFromLPS_CXYZT(TBuffer * buffer, size_t size)
+{
+  int numberOfVectors = size / 3;
+  for (size_t i = 0; i < numberOfVectors; ++i)
+  {
+    buffer[0] = -buffer[0];
+    buffer[1] = -buffer[1];
+    buffer += 3;
+  }
+}
+
+// Internal function to convert vectors between RAS and LPS coordinate systems.
+// Dimensions are XYZTC (NIFTI memory layout)
+template <typename TBuffer>
+void
+ConvertRASToFromLPS_XYZTC(TBuffer * buffer, size_t size)
+{
+  // Flip the sign of the first two components (L<->R, P<->A)
+  // and keep the third component (S) unchanged.
+  int numberOfVectors = size / 3 * 2;
```

Comment:
In this context, I think `numberOfVectors` is not really the number of vectors, right? It seems to me that it is the number of components whose sign is to be flipped. If so, maybe rename it to `numberOfComponents`...?

Reply:
Correct. numberOfVectors=size/3. The numberOfVectors*2 is the number of components we update.

Reply:
Thanks! Don't forget the `int` here as well!